### PR TITLE
[FIX] Invalid filters on the Omnichannel Analytics page

### PR DESCRIPTION
**Pull request by murtaza98**
## Issue(s)

1. Fix layout not rendering properly. Following was the layout which was getting rendered 
    **BEFORE**
    image
    **AFTER**
    image

2. Handle some edge cases for date select
image

- When we select `This Month`, then it should set the end date to **current date** instead of the last date of the month
- Earlier when we were using [`date.toISOString()`](https://github.com/RocketChat/Rocket.Chat/blob/5c23deecbe95d2cfec1f76f30ede3fb6af0dfc64/client/views/omnichannel/analytics/DateRangePicker.js#L9), under my timezone it was converting start of the month incorrectly. I expected it to be 1st December however it converts it to 30st November. Please have a look in the below console
image
To solve :top: error, I've converted the code to use `moment` library.

## Further comments

Server version: In sync with latest develop branch

### CODE REVIEW

**Review by rafaelblink — APPROVED** [review]
Everything working well.

**Comment by renatobecker** [comment]
Screen Shot 2020-12-21 at 18 40 36

@murtaza98 this is the design expected.

**Review by renatobecker — CHANGES_REQUESTED** [review]
Screen Shot 2020-12-21 at 18 40 36

@murtaza98 this is the design expected.

```
I20201221-18:35:31.564(-3)? Incorrect chart options
I20201221-18:35:31.571(-3)? livechat:getAnalyticsOverviewData => Invalid dates
I20201221-18:35:31.576(-3)? Incorrect analytics options
```

File: client/views/omnichannel/analytics/InterchangeableChart.js [hunk 1]
```diff
@@ -24,11 +24,13 @@ const InterchangeableChart = ({ departmentId, dateRange, chartName, ...props })
 
 	const draw = useMutableCallback(async (params) => {
 		try {
-			const result = await loadData(params);
-			if (!(result && result.chartLabel && result.dataLabels && result.dataPoints)) {
-				return console.log('livechat:getAnalyticsChartData => Missing Data');
+			if (params.daterange.from && params.daterange.to) {
```
Comment (renatobecker) [inline client/views/omnichannel/analytics/InterchangeableChart.js:27]:
```suggestion
			if (!params?.daterange?.from || !params?.daterange?.to) {
				return;
			}
				
			const result = await loadData(params);
			if (!result?.chartLabel || !result?.dataLabels !! !result?.dataPoints)) {
				return console.log('livechat:getAnalyticsChartData => Missing Data');
			}

			context.current = await drawLineChart(canvas.current, context.current, [result.chartLabel], result.dataLabels, [result.dataPoints]);

```

File: client/views/omnichannel/analytics/InterchangeableChart.js [hunk 2]
```diff
@@ -24,11 +24,13 @@ const InterchangeableChart = ({ departmentId, dateRange, chartName, ...props })
...
+				const result = await loadData(params);
+				if (!(result && result.chartLabel && result.dataLabels && result.dataPoints)) {
+					return console.log('livechat:getAnalyticsChartData => Missing Data');
```
Comment (renatobecker) [inline client/views/omnichannel/analytics/InterchangeableChart.js:30]: There is no reason to display this message on the console log, if something went wrong we should alert the user via notification.

**Review by renatobecker — CHANGES_REQUESTED** [review]

**Comment by renatobecker** [comment]
@murtaza98 I'm still facing this inconsistency

Screen Shot 2021-01-14 at 09 22 10

**Comment by murtaza98** [comment]
> @murtaza98 I'm still facing this inconsistency
> 
> Screen Shot 2021-01-14 at 09 22 10

@renatobecker This bug hasn't been fixed yet. I wasn't able to determine the issue :disappointed: . I'll still need to do some more debugging specifically with mongo queries.

**Review by renatobecker — APPROVED** [review]

**Pull request merged**